pkg/olm/clusterserviceversion_test.go

```diff
@@ -0,0 +1,349 @@
+			expectedError: fmt.Errorf("alm-examples not found in given clusterserviceversion named clusterservice"),
+		},
+		{
+			almExample:     map[string]string{"alm-examples": "test"},
+			clusterService: buildValidClusterServiceBuilder(buildTestClientWithDummyClusterServiceObject()),
+			expectedError:  fmt.Errorf("alm-examples not found in given clusterserviceversion named clusterservice"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		almExamples, err := testCase.clusterService.GetAlmExamples()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, almExamples, testCase.almExample["alm-examples"])
+		}
+	}
+}
+
+func TestClusterServiceGetPhase(t *testing.T) {
+	testCases := []struct {
+		clusterService *ClusterServiceVersionBuilder
+		expectedError  error
+		phase          oplmV1alpha1.ClusterServiceVersionPhase
+	}{
+		{
+			clusterService: buildValidClusterServiceBuilder(
+				buildTestClientWithDummyClusterServiceObjectWitPhase(oplmV1alpha1.CSVPhaseSucceeded)),
+			phase:         oplmV1alpha1.CSVPhaseSucceeded,
+			expectedError: nil,
+		},
+		{
+			clusterService: buildValidClusterServiceBuilder(
+				clients.GetTestClients(clients.TestClientParams{SchemeAttachers: testSchemes})),
+			phase:         oplmV1alpha1.CSVPhaseSucceeded,
+			expectedError: fmt.Errorf("clusterservice clusterserviceversion not found in test-namespace namespace"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		phase, err := testCase.clusterService.GetPhase()
+		assert.Equal(t, testCase.expectedError, err)
+
+		if testCase.expectedError == nil {
+			assert.Equal(t, testCase.phase, phase)
+		}
+	}
+}
+
+func TestClusterServiceIsSuccessful(t *testing.T) {
+	testCases := []struct {
+		clusterService *ClusterServiceVersionBuilder
+		expectedError  error
+		successful     bool
+	}{
+		{
+			clusterService: buildValidClusterServiceBuilder(
+				buildTestClientWithDummyClusterServiceObjectWitPhase(oplmV1alpha1.CSVPhaseSucceeded)),
+			successful:    true,
+			expectedError: nil,
+		},
+		{
+			clusterService: buildValidClusterServiceBuilder(
+				buildTestClientWithDummyClusterServiceObjectWitPhase(oplmV1alpha1.CSVPhaseFailed)),
+			successful:    false,
+			expectedError: nil,
+		},
+		{
+			clusterService: buildValidClusterServiceBuilder(
+				clients.GetTestClients(clients.TestClientParams{SchemeAttachers: testSchemes})),
+			successful: false,
+			expectedError: fmt.Errorf("failed to get phase value for clusterservice clusterserviceversion in " +
+				"test-namespace namespace due to clusterservice clusterserviceversion not found in test-namespace namespace"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		successful, err := testCase.clusterService.IsSuccessful()
+		if testCase.expectedError != nil {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		} else {
+			assert.Nil(t, err)
+			assert.Equal(t, testCase.successful, successful)
+		}
```

Comment:
```suggestion
		assert.Equal(t, testCase.expectedError, err)
		
		if testCase.expectedError == nil {
			assert.Equal(t, testCase.successful, successful)
		}
```

Reply:
The return error can be tricky and I need to convert it to string. The code that you proposed didn't work.